### DISCUSSION TITLE
Custom download strategy

### DISCUSSION THREAD

**Discussion by jonnybbb** [jreleaser/jreleaser#1554] [Q&A]
I would like to use a custom download strategy for downloading the artefacts. 

```
require_relative "lib/private"
.....
  if OS.linux? && Hardware::CPU.intel?
    URL "<url>", using: GitHubPrivateRepositoryReleaseDownloadStrategy
    sha256 "<sha>"
  end
  if OS.mac? && Hardware::CPU.arm?
    url "<url>", using: GitHubPrivateRepositoryReleaseDownloadStrategy
    sha256 "<sha>"
  end
```
Is there a way to configure this in jReleaser? 

goreleaser [has this](https://goreleaser.com/customization/homebrew/):
```
    # Allows you to set a custom download strategy. Note that you'll need
    # to implement the strategy and add it to your tap repository.
    # Example: https://docs.brew.sh/Formula-Cookbook#specifying-the-download-strategy-explicitly
    download_strategy: CurlDownloadStrategy

    # Allows you to add a custom require_relative at the top of the formula
    # template.
    custom_require: custom_download_strategy
```

**Comment by aalmiray** [comment]
We don't support custom download strategies at the moment, you're the first one asking for such behavior 😅

I suggest you to setup a custom brew template with additional files for the time being as a workaround.

Please feel free to raise a ticket for this feature 😀

**Reply by aalmiray** [reply]
FWIW I think a custom template for handling custom download strategies in a multi-platform formula won't work at the moment.

**Reply by aalmiray** [reply]
Filed https://github.com/jreleaser/jreleaser/issues/1555
